src/opm/input/eclipse/Schedule/Schedule.cpp

```diff
@@ -1276,8 +1277,19 @@ File {} line {}.)", pattern, location.keyword, location.filename, location.linen
     }
 
     double Schedule::stepLength(std::size_t timeStep) const {
-        auto elapsed = this->snapshots[timeStep].end_time() - this->snapshots[timeStep].start_time();
-        return std::chrono::duration_cast<std::chrono::seconds>(elapsed).count();
+        const auto start = this->snapshots[timeStep].start_time();
+        const auto end = this->snapshots[timeStep].end_time();
+        if (start > end) {
+            throw std::invalid_argument {
+                    fmt::format(" Report step {} has start time after end time,\n"
+                                "   * Start time = {:%d-%b-%Y %H:%M:%S}\n"
+                                "   * End time   = {:%d-%b-%Y %H:%M:%S}.\n"
+                                " Possibly due to inconsistent RESTART/SKIPREST settings.",
+                                timeStep + 1,
+                                fmt::gmtime(TimeService::to_time_t(start)),
+                                fmt::gmtime(TimeService::to_time_t(end))) };
+        };
```

Comment:
Tiny nit, but a trailing `;` on the closing brace of `if()`, while not an error, does look a little weird here.